src/main/scala/com/microsoft/hyperspace/Hyperspace.scala

```diff
@@ -163,6 +163,17 @@ class Hyperspace(spark: SparkSession) {
   def index(indexName: String): DataFrame = {
     indexManager.index(indexName)
   }
+
+  /**
+   * Get index metadata and detailed index statistics for a given index and log version.
+   *
+   * @param indexName Name of the index to get stats for.
+   * @param logVersion Index log version to retrieve.
+   * @return Index metadata and statistics as a [[DataFrame]].
+   */
+  def index(indexName: String, logVersion: Int): DataFrame = {
```

Comment:
https://github.com/microsoft/hyperspace/pull/272#discussion_r598329940 was actually asking to add an API to `IndexManager` to return available index versions. For example, `def getAvailableIndexVersions(indexName: String): Seq[Int]` so that versions can be used for `def getIndex(indexName: String, logVersion: Int)`. Does it make sense?

We can also update the existing `def index(indexName: String): DataFrame` to return all available versions of index instead of exposing a new `def index(indexName: String, logVersion: Int)` (I don't think we need to expose this as public). Btw, you don't have to address this in this PR since I am not sure if we need it or not. We can just concentrate on updating `IndexLogManager` in this PR.

Reply:
>#272 (comment) was actually asking to add an API to IndexManager to return available index versions. For example, def getAvailableIndexVersions(indexName: String): Seq[Int] so that versions can be used for def getIndex(indexName: String, logVersion: Int). Does it make sense?

Yep then we could fix #387 with the API. I will update this PR.

>We can also update the existing def index(indexName: String): DataFrame to return all available versions of index instead of exposing a new def index(indexName: String, logVersion: Int) (I don't think we need to expose this as public). Btw, you don't have to address this in this PR since I am not sure if we need it or not. We can just concentrate on updating IndexLogManager in this PR.

I think it might be good to expose this API since we expose the log version in plan. A user might want to check the used log version.

Reply:
> I think it might be good to expose this API since we expose the log version in plan. A user might want to check the used log version.

If we want to expose the log versions, we can change `index` to return log entries of multiple versions (add an optional param to return last N entries). Since we return `DataFrame`, the user can just filter by the version if needed; I don't think it's necessary to expose this new API, which will require us to expose another API to return available versions.

Reply:
@imback82 BTW, currently if an index is deleted & vacuumed and then a new index with the same index name is created, log files are not removed & the log number is continued. I think this can be a problem. WDYT?

Reply:
Good point. It was intentional to keep the log files around (compliance reason), but I think we could provide an option to remove log files.

Reply:
Feel free to create an issue so we can track it. Thanks.